Sources_v3/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -0,0 +1,203 @@
+        stackView.topAnchor.constraint(greaterThanOrEqualTo: containerCenterView.topAnchor, constant: 0).isActive = true
+        stackView.bottomAnchor.constraint(lessThanOrEqualTo: containerCenterView.bottomAnchor, constant: 0).isActive = true
+        stackView.pin(anchors: [.leading, .trailing, .centerY], to: containerCenterView)
+        
+        // Top part of centerStackView.
+        let topCenterStackView = UIStackView()
+        topCenterStackView.alignment = .top
+        topCenterStackView.addArrangedSubview(titleLabel)
+        topCenterStackView.addArrangedSubview(unreadCountBubble)
+        
+        // Bottom part of centerStackView.
+        let bottomCenterStackView = UIStackView()
+        bottomCenterStackView.spacing = UIStackView.spacingUseSystem
+        
+        subtitleLabel.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        subtitleLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+        timestampLabel.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        
+        bottomCenterStackView.addArrangedSubview(subtitleLabel)
+        bottomCenterStackView.addArrangedSubview(readStatusView)
+        bottomCenterStackView.addArrangedSubview(timestampLabel)
+        
+        stackView.addArrangedSubview(topCenterStackView)
+        stackView.addArrangedSubview(bottomCenterStackView)
+        
+        container.centerStackView.isHidden = false
+        container.centerStackView.addArrangedSubview(containerCenterView)
+    
+        avatarView.widthAnchor.constraint(equalToConstant: 56).isActive = true
+    }
+    
+    open func updateContent() {
+        guard let channel = channel else {
+            resetContent()
+            return
+        }
+        
+        // Title
+        
+        titleLabel.text = channel.displayName
+        
+        // Subtitle
+        
+        if let typingMembersInfo = typingMemberString {
+            subtitleLabel.text = typingMembersInfo
+        } else if let latestMessage = channel.latestMessages.first {
+            subtitleLabel.text = "\(latestMessage.author.displayName): \(latestMessage.text)"
+        } else {
+            subtitleLabel.text = "No messages"
+        }
+        
+        // Avatar
+        
+        avatarView.channel = channel
+        
+        // UnreadCount
+        
+        // Mock test code
+        unreadCountBubble.unreadCount = channel.unreadCount
+        unreadCountBubble.invalidateIntrinsicContentSize()
+        
+        // Timestamp
+        
+        timestampLabel.text = channel.lastMessageAt?.getFormattedDate(format: "hh:mm a")
+        
+        // TODO: ReadStatusView
+        // Missing LLC API
+    }
+    
+    open func resetContent() {
+        titleLabel.text = ""
+        subtitleLabel.text = ""
+        avatarView.channel = nil
+        unreadCountBubble.unreadCount = .noUnread
+        timestampLabel.text = ""
+        readStatusView.status = .empty
+    }
```

Comment:
Can we reset the content inside `updateContent`?

And instead of:
```swift
guard let channel = channel else {
  resetContent()
  return
}
titleLabel.text = channel.displayName
```
have just:
```swift
titleLabel.text = channel?.displayName
```

Reply:
Yeah, we can, but we have `ChatUnreadCountBubble` that accepts non-nil `ChannelUnreadCount`, also extra effort will be needed for resetting `subtitle`. And logic of the resetting will be not obvious from the glance cause will be in different places. So not sure if it's better idea.
WDYT?

Reply:
That's not a big from my perspective (maybe I'm missing smth). Will that work?

```swift
subtitleLabel.text = channel?.typingMembersOrLastMessage
unreadCountBubble.unreadCount = channel?.unreadCount ?? . noUnread

extension Channel {
 var typingMembersOrLastMessage: String {
  //if statements go here
 }
}
```

Reply:
Yep, that should work, just not that obvious that stuff is reseted. 
I'm not strong about that though :)

Reply:
I think that's great to have only 1 place where content is update. End-users will also appreciate it :)